src/main/java/com/mcmoddev/lib/util/TabContainer.java

```diff
@@ -1,17 +1,122 @@
+ *
+ */
+@Deprecated
+public final class TabContainer implements ITabProvider {
+	// @SkyBlade: temporary code until the tabcontainer is replaced
+	
+	private static final String TAB_NOT_FOUND = "Tab not found: ";
 	public final MMDCreativeTab blocksTab;
 	public final MMDCreativeTab itemsTab;
 	public final MMDCreativeTab toolsTab;
 
+	private List<Tuple2<String, String>> tabItemMapping;
+	
 	public TabContainer(MMDCreativeTab blocksTab, MMDCreativeTab itemsTab, MMDCreativeTab toolsTab) {
 		this.blocksTab = blocksTab;
 		this.itemsTab = itemsTab;
 		this.toolsTab = toolsTab;
+		
+		tabItemMapping = new ArrayList<>();
 	}
 
-}
+	public TabContainer(MMDCreativeTab blocksTab, MMDCreativeTab itemsTab, MMDCreativeTab toolsTab, List<Tuple2<String, String>> tabItemMapping) {
+		this.blocksTab = blocksTab;
+		this.itemsTab = itemsTab;
+		this.toolsTab = toolsTab;
+		this.tabItemMapping = tabItemMapping;
+	}
+	
+	@Override
+	public MMDCreativeTab getTabByName(String tabName) throws TabNotFoundException {
+		switch (tabName) {
+		case "blocksTab":
+			return this.blocksTab;			
+		case "itemsTab":
+			return this.itemsTab;
+		case "toolsTab":
+			return this.toolsTab;
+		default:
+			throw new TabNotFoundException(tabName);
+		}
+	}
+
+	@Override
+	public void addBlockToTab(String tabName, Block block) throws TabNotFoundException {
+		MMDCreativeTab tab = this.getTabByName(tabName);
+		
+		if (tab == null) 
+			throw new TabNotFoundException(TAB_NOT_FOUND + tabName);
+		
+		block.setCreativeTab(tab);
+	}
+	
+	@Override
+	public void addItemToTab(String tabName, Item item) throws TabNotFoundException {
+		MMDCreativeTab tab = this.getTabByName(tabName);
+		
+		if (tab == null) 
+			throw new TabNotFoundException(TAB_NOT_FOUND + tabName);
+		
+		item.setCreativeTab(tab);
+	}
+
+	@Override
+	public void setIcon(String tabName, String materialName) throws TabNotFoundException, MaterialNotFoundException {
+		MMDCreativeTab tab = this.getTabByName(tabName);
+		
+		if (tab == null) 
+			throw new TabNotFoundException(TAB_NOT_FOUND + tabName);
+		
+		Block temp;
+		ItemStack blocksTabIconItem;
+
+		MMDMaterial material = Materials.getMaterialByName(materialName);
+		
+		if (material.getName() == materialName && (material.hasBlock(Names.BLOCK)))
+			temp = material.getBlock(Names.BLOCK);
+		else
+			temp = net.minecraft.init.Blocks.IRON_BLOCK;
+		
+		blocksTabIconItem = new ItemStack(Item.getItemFromBlock(temp));
+		blocksTab.setTabIconItem(blocksTabIconItem);
+	}
+
+	@Override
+	public String getTab(String itemName, String modID) {
+		return getTab(itemName);
+	}
+
+	@Override
+	public void setTabItemMapping(String tabName, String itemName) {
+		tabItemMapping.add(new Tuple2<String, String>(tabName, itemName));
+	}
+
+	@Override
+	public String getTab(String itemName) {
+		List<Tuple2<String, String>> matchingTabs =  tabItemMapping.stream().filter(entry -> entry.y == itemName).collect(Collectors.toList());
+		
+		if (matchingTabs.isEmpty())

```

Comment:
Instead of collecting to a list for `isEmpty`, I recommend:
```java
if (tabItemMapping.stream().filter(entry -> entry.y.equals(itemName)).findAny().isPresent()) {
    return "blocksTab"
}
return "";
```

Reply:
Ahh I was being an idiot/was coding tired this logic is wrong it's supposed to default to blocksTab if it doesn't find the one being searched for, otherwise return the tab found in the mapping (its deprecated code anyway but it should at least be correct), I'm fixing it now